### PR TITLE
Parse table and column stats for hive

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

Parse table and column statistics for hive tables.

### 🤓 What?

- Added logic for parsing table and column stats.
- Added config value for collecting these stats.
- Unit test.

### 🧪 Tested?

Unit test.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

**Review by mars-lan — COMMENTED** [review]
Please also update README with requirements on privileges.

File: metaphor/hive/README.md [hunk 1]
```diff
@@ -13,8 +13,16 @@ host: <host>
 port: <port>
 auth_user: <auth user for hiveserver>
 password: <password for the auth user>
+
+output:
+  file:
+    directory: <output_directory>
 ```
 
+If `auth_user` and `password` are provided, ensure the specified user has `SELECT` privilege. See [https://cwiki.apache.org/confluence/display/Hive/SQL+Standard+Based+Hive+Authorization](https://cwiki.apache.org/confluence/display/Hive/SQL+Standard+Based+Hive+Authorization) for more information.
```
Comment (mars-lan) [inline metaphor/hive/README.md:22]: Add a `Setup` section like others. Also is do you need `SELECT` privilege for ALL tables, or some. Do you need that even for SHOW & DESCRIBE?

Comment (usefulalgorithm) [reply metaphor/hive/README.md:22]: >  Do you need that even for SHOW & DESCRIBE?

Yes, see the table in the `Privileges Required for Hive Operations` section in the linked page.

Comment (mars-lan) [reply metaphor/hive/README.md:22]: Alos use `SQL Standard Based Hive Authorization` as the title of the link.

Comment (mars-lan) [reply metaphor/hive/README.md]: Looks like we also need `WITH GRANT OPTION`: https://cwiki.apache.org/confluence/display/Hive/SQL+Standard+Based+Hive+Authorization#:~:text=Y%20(input%20location)-,SHOW%20CREATE%20TABLE,-Y%2BG

File: metaphor/hive/README.md [hunk 2]
```diff
@@ -11,10 +15,31 @@ Create a YAML config file based on the following template.
...
+```
+
+See [Output Config](../common/docs/output.md) for more information on `output`.
+
+### Optional Configurations
+
+#### Authenticated User
+
+```yaml
 auth_user: <auth user for hiveserver>
```
Comment (mars-lan) [inline metaphor/hive/README.md:31]: Chances are these will be required for any reasonable production deployment?

File: metaphor/hive/extractor.py [hunk 3]
```diff
@@ -85,8 +95,87 @@ def _extract_table(
...
+                data_size_bytes=table_size,
+                record_count=num_rows,
+            )
+            field_statistics: List[FieldStatistics] = []
+            for field in fields:
+                cursor.execute(
+                    f"describe formatted {database}.{table} {field.field_path}"
+                )
+                stats_col_names = {
+                    "num_nulls": "nullValueCount",
```
Comment (mars-lan) [inline metaphor/hive/extractor.py:141]: What bout avg & std dev (which are expensive to compute)? Maybe we should have a config like https://github.com/MetaphorData/connectors/blob/main/metaphor/common/docs/column_statistics.md?

Comment (usefulalgorithm) [reply metaphor/hive/extractor.py:141]: Hive does not compute those AFAIK. Don't think we want to compute these by hand?

File: metaphor/hive/README.md (hunk 2 above)
Comment (usefulalgorithm) [reply metaphor/hive/README.md:31]: Yes. How should I write this?

File: metaphor/hive/extractor.py (hunk 3 above)
Comment (usefulalgorithm) [reply metaphor/hive/extractor.py:155]: There are indeed functions that calculate avg and std. Added those and checked that they work.

Comment (mars-lan) [reply metaphor/hive/extractor.py:155]: std & avg are expensive to calculate so I'd suggest giving users the choice to selectively enable each like https://github.com/MetaphorData/connectors/tree/main/metaphor/snowflake/profile#column-statistics

**Review by mars-lan — APPROVED** [review]

**Pull request merged**